CONTRIBUTING.md

```diff
@@ -3,104 +3,190 @@
-
-For new code, please run [`clint.py`][clint] to detect style errors. It is not
-perfect and may have false positives and negatives. To have `clint.py` ignore
-certain special cases, put `// NOLINT` at the end of the line.
-
-We also provide a configuration file for [`clang-format` and 
-`git-clang-format`][clang-format], which can be used to format code according
-to the style guidelines. Be aware this formatting method might need user
-supervision.
-
-#### Commit guidelines
-
-The purpose of these guidelines is to *make reviews easier* and make the VCS logs more valuable.
-
-- Try to keep the first line under 70 characters.
-- Include further description, if necessary, after a blank line.
-    - Don't make it too verbose by including obvious things.
-    - But don't spare clarifications for anything that could be not so obvious.
-      Some commit messages are pages long, and that's fine if there's no better
-      place for those comments to live.
+We have a [style guide][style-guide] that all new code should follow.
+However, large portions of the existing Vim codebase violate it to some
+degree, and fixing them would increase merge conflicts and add noise to `git
+blame`.
+
+Weigh those costs when making cosmetic changes. In general, avoid pull
+requests dominated by style changes, but feel free to fix up lines that you
+happen to be modifying anyway. Fix anything that looks outright
+[barbarous](http://www.orwell.ru/library/essays/politics/english/e_polit), but
+otherwise prefer to leave things as they are.
+
+For new code, run `make lint` (which runs [clint.py][clint]) to detect style
+errors. Make sure that the file(s) you intend to be linted are not in
+`clint-ignored-files.txt`. It's not perfect, so some warnings may be false
+positives/negatives. To have `clint.py` ignore certain cases, put `// NOLINT`
+at the end of the line.
+
+We also provide a configuration file for [`clang-format`][clang-format], which
+can be used to format code according to the style guidelines. Be aware that
+this formatting method might need user supervision. To have `clang-format`
+ignore certain line ranges, use the following special comments:
+
+```c
+int formatted_code;
+// clang-format off
+    void    unformatted_code  ;
+// clang-format on
+    void formatted_code_again;
+```
+
+### Commit guidelines
+
+The purpose of these guidelines is to *make reviews easier* and make the
+[VCS][vcs] logs more valuable.
+
+- Try to keep the first line under 72 characters.
+- If necessary, Include further description after a blank line.
+    - Don't make the description too verbose by including obvious things, but
+      don't spare clarifications for anything that may be not so obvious.
+      Some commit messages are pages long, and that's fine if there's no
+      better place for those comments to live.
     - **Recommended:** Prefix logically-related commits with a consistent
-      identifier at the beginning of each commit message.
```

Comment:
Add "For already used identifiers, have a look a the commit history of the respective file/code parts" or something like that?